sdk/storage/storage-blob/src/pollers/BlobStartCopyFromUrlPoller.ts

```diff
@@ -0,0 +1,187 @@
+import { BlobStartCopyFromURLResponse } from "../generated/src/models";
+import { BlobClient, BlobStartCopyFromURLOptions } from "../BlobClient";
+
+/**
+ * Defines the operations needed from a BlobClient.
+ */
+export type CopyPollerBlobClient = Pick<BlobClient, "abortCopyFromURL" | "getProperties"> & {
+  startCopyFromURL(
+    copySource: string,
+    options?: BlobStartCopyFromURLOptions
+  ): Promise<BlobStartCopyFromURLResponse>;
+};
+
+/**
+ * The state used by the PollOperation.
+ */
+export interface BlobBeginCopyFromUrlPollState
+  extends PollOperationState<BlobStartCopyFromURLResponse> {
+  blobClient: CopyPollerBlobClient;
+  copyId?: string;
+  copyProgress?: string;
+  copySource: string;
+  startCopyFromURLOptions?: BlobStartCopyFromURLOptions;
+}
+
+/**
+ * The PollOperation responsible for:
+ *  - performing the initial startCopyFromURL
+ *  - checking the copy status via getProperties
+ *  - cancellation via abortCopyFromURL
+ */
+export interface BlobBeginCopyFromURLPollOperation
+  extends PollOperation<BlobBeginCopyFromUrlPollState, BlobStartCopyFromURLResponse> {}
+
+/**
+ *
+ */
+export interface BlobBeginCopyFromUrlPollerOptions {
+  blobClient: CopyPollerBlobClient;
+  copySource: string;
+  intervalInMs?: number;
+  onProgress?: (state: BlobBeginCopyFromUrlPollState) => void;
+  resumeFrom?: string;
+  startCopyFromURLOptions?: BlobStartCopyFromURLOptions;
+}
+
+/**
+ *
+ */
+export class BlobBeginCopyFromUrlPoller extends Poller<
+  BlobBeginCopyFromUrlPollState,
+  BlobStartCopyFromURLResponse
+> {
+  public intervalInMs: number;
+
+  constructor(options: BlobBeginCopyFromUrlPollerOptions) {
+    const {
+      blobClient,
+      copySource,
+      intervalInMs = 15000,
+      onProgress,
+      resumeFrom,
+      startCopyFromURLOptions
+    } = options;
+
+    let state: BlobBeginCopyFromUrlPollState | undefined;
+
+    if (resumeFrom) {
+      state = JSON.parse(resumeFrom).state;
+    }
+
+    const operation = makeBlobBeginCopyFromURLPollOperation({
+      ...state,
+      blobClient,
+      copySource,
+      startCopyFromURLOptions
+    });
+
+    super(operation);
+
+    if (typeof onProgress === "function") {
+      this.onProgress(onProgress);
+    }
+
+    this.intervalInMs = intervalInMs;
+  }
+
+  async delay(): Promise<void> {
+    return delay(this.intervalInMs);
+  }
+}
+
+const cancel: BlobBeginCopyFromURLPollOperation["cancel"] = async function cancel(
+  this: BlobBeginCopyFromURLPollOperation,
+  options = {}
+) {
+  const state = this.state;
+  const { copyId } = state;
+  if (state.isCompleted) {
+    return makeBlobBeginCopyFromURLPollOperation(state);
+  }
+
+  if (!copyId) {
+    state.isCancelled = true;
+    return makeBlobBeginCopyFromURLPollOperation(state);
+  }
+
+  try {
+    await state.blobClient.abortCopyFromURL(copyId, {
+      abortSignal: options.abortSignal
+    });
+    state.isCancelled = true;
+  } catch (err) {
+    // swallow error
```

Comment:
>// swallow error [](start = 4, length = 16)

Why should swallow the error?

What if Storage server returns 500 ServerInternalError?

Reply:
In this case the abort failed, so the poller continues. I'm not sure how we should surface an error cancelling the operation to the user, but will look into it.

Reply:
Updated to throw the error. This propagates correctly to the cancelOperation call.